libsplinter/src/protocol/network.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2018-2021 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::protos::network;
+use crate::protos::prelude::*;
+
+/// The network message envelope
+#[derive(Debug)]
+pub enum NetworkMessage {
+    NetworkEcho(NetworkEcho),
+    NetworkHeartbeat(NetworkHeartbeat),
+    Circuit(Vec<u8>),
```

Comment:
Is this `Vec<u8>` because we don't have the protocol structs for Circuit messages yet?

Reply:
Yes, once the circuit ones are defined the Vec<u8> will be replaced with the CircuitMessage protocol.